tests/e2e/utils/frontend/frontend-utils.page.ts

```diff
@@ -32,21 +27,47 @@ export class FrontendUtils {
 			await this.page
 				.getByLabel( `Add “${ itemName }” to your cart` )
 				.click();
-			await this.page.waitForResponse( /add_to_cart/ );
+			await this.page.waitForResponse( /add_to_cart|batch/ );
 			return;
 		}
 		await this.page.click( 'text=Add to cart' );
 	}
 
+	async goToCheckout() {
+		await this.page.goto( '/checkout', {
+			waitUntil: 'commit',
+		} );
+	}
+
 	async goToShop() {
 		await this.page.goto( '/shop', {
 			waitUntil: 'commit',
 		} );
 	}
 
-	async goToCheckout() {
-		await this.page.goto( '/checkout', {
-			waitUntil: 'commit',
+	async emptyCart() {
+		await this.page.goto( '/cart', { waitUntil: 'domcontentloaded' } );
+		await this.page.evaluate( () => {
+			const store = window?.wp?.data?.select( 'wc/store/cart' );
+			if ( ! store ) {
+				return new Error(
+					'You must be on a page with data stores before using frontendUtils.emptyCart.'
+				);
+			}
+			const cartData = store.getCartData();
+			if ( ! Array.isArray( cartData?.items ) ) {
+				return new Error(
+					'cartData.items must be an array. If it is not'
```

Comment:
"cartData.items must be an array. If it is not" is this missing something?

Reply:
Removed now as it's been updated to use API, but yes I left that in by accident it should have just finished after `cartData.items must be an array.`